src/itype.h

```diff
@@ -763,6 +763,17 @@ struct islot_artifact {
     int dream_freq_met;
 };
 
+// A name that is applied under certain conditions.
+struct conditional_name {
+    // Context type  (i.e. "FLAG"          or "COMPONENT_ID")
+    std::string type;
```

Comment:
It seems one can only use certain type strings (e.g. a type "foobar" would not be recognised by the game). However, it is not clear what values are supported. And even the loading code does not check for supported values.

This member should be an enumeration. Loading the JSON data should convert the string from JSON into an enumeration value. This allows to check for correctness at *load* time. And it clearly lists what values are possible.

Reply:
I changed it to an enum, but I'm not totally sure it's checking for correctness the way you were suggesting. A re-review would be appreciated, if you have the time.